app/assets/javascripts/pretty-text/addon/sanitizer.js

```diff
@@ -77,7 +77,7 @@ export function sanitize(text, allowLister) {
             (forAttr.indexOf("*") !== -1 || forAttr.indexOf(value) !== -1)) ||
           (name.indexOf("data-html-") === -1 &&
             name.indexOf("data-") === 0 &&
-            forTag["data-*"]) ||
+            Object.keys(forTag).find((k) => k.match(/data-.*-?\*/))) ||
```

Comment:
I'm not sure if I'm reading this right... but I think this would still allow all `data-` attributes, even if you only specify `pre[data-code-*]` in the sanitizer?

To be sure, could we add a test which checks that `<code data-some-malicious-key="value"` is blocked?

Reply:
Yes right, that should be better, I also feel like we are actually not check for value in this code. I'm not going to merge this now anyways, will think about it and finish Monday, thanks :+1:

Reply:
@davidtaylorhq  will read it again after Lunch, but that should improve things: https://github.com/discourse/discourse/pull/15743/commits/0a0cbf1c45997b8fcd2d48e7ad2f549695299f94